code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -26,20 +26,16 @@ import com.adobe.marketing.mobile.notificationbuilder.internal.util.Notification
  * This class is used to parse the push template data payload or an intent and provide the necessary information
  * to build a notification.
  */
-internal sealed class AEPPushTemplate(data: NotificationData) {
-
-    // Message data payload for the push template
-    internal lateinit var messageData: MutableMap<String, String>
-        private set
+internal sealed class AEPPushTemplate(val data: NotificationData) {
 
     // Required, title of the message shown in the collapsed and expanded push template layouts
-    internal val title: String
+    internal val title: String = data.getRequiredString(PushPayloadKeys.TITLE)
 
     // Required, body of the message shown in the collapsed push template layout
-    internal val body: String
+    internal val body: String = data.getRequiredString(PushPayloadKeys.BODY)
 
     // Required, Version of the payload assigned by the authoring UI.
-    internal val payloadVersion: String
+    internal val payloadVersion: String = data.getRequiredString(PushPayloadKeys.VERSION)
```

Comment:
Can you initialize these values in the `init` block where the other member are done.